### PR TITLE
Clean up to the item buy/sell checks

### PR DESCRIPTION
**Pull request by aleos89**
* **Addressed Issue(s)**: N/A

* **Server Mode**: Pre-renewal and Renewal

* **Description of Pull Request**: 
  * An item's sell value will now always be calculated whether the item exists or not. This resolves adjustment of an item's buy value during the import process.
  * Moves a couple checks to the loadingFinished function.
Thanks to @Daegaladh!

### CODE REVIEW

**Review by Lemongrass3110 — APPROVED** [review]

File: src/map/itemdb.cpp [hunk 1]
```diff
@@ -157,15 +148,7 @@ uint64 ItemDatabase::parseBodyNode(const YAML::Node &node) {
 		item->value_buy = buy;
```
Comment (Atemo) [inline src/map/itemdb.cpp:148]:
```suggestion
		item->value_buy = buy;
		item->value_sell = 0;
```
Define `buy` should 'remove' the current `sell`

File: src/map/itemdb.cpp [hunk 2]
```diff
@@ -177,13 +160,9 @@ uint64 ItemDatabase::parseBodyNode(const YAML::Node &node) {
 
 		item->value_sell = sell;
```
Comment (Atemo) [inline src/map/itemdb.cpp:161]:
```suggestion
		item->value_sell = sell;
		item->value_buy = 0;
```
Define `sell` should 'remove' the current `buy`

**Review by Atemo — APPROVED** [review]

File: src/map/itemdb.cpp (hunk 1 above)
Comment (Lemongrass3110) [reply src/map/itemdb.cpp:148]: @Atemo only if the entry already exists, correct? Applies to both of your comments.

Comment (Atemo) [reply src/map/itemdb.cpp:148]: Well yes you're right the values are re-assigned when the entry doesn't exist but it doesn't change the result it's still correct

**Pull request merged**